.github/workflows/docker-build-push-image.yml

```diff
@@ -0,0 +1,109 @@
+name: Build and Push Docker image
+
+on:
+  workflow_call:
+    inputs:
+      aws_ecr:
+        description: "Push to AWS ECR"
+        default: true
+        required: false
+        type: boolean
+      docker_hub:
+        description: "Push to Docker Hub"
+        default: true
+        required: false
+        type: boolean
+    secrets:
+      AWS_ACCOUNT_ID:
+        description: "The AWS account ID used to determine the ECR registry"
+        required: true
+      AWS_REGION:
+        description: "The AWS region used to determine the ECR registry"
+        required: true
+      AWS_ECR_ACCESS_KEY_ID:
+        description: "The access key ID used to log into AWS ECR"
+        required: true
+      AWS_ECR_SECRET_ACCESS_KEY:
+        description: "The secret access key ID used to log into AWS ECR"
+        required: true
+      DOCKERHUB_USERNAME:
+        description: "The username used to log into Docker Hub"
+        required: true
+      DOCKERHUB_PASSWORD:
+        description: "The password used to log into Docker Hub"
+        required: true
+
+permissions:
+  contents: write
```

Comment:
I need to use `write` here due to the Docker layer caching being written to cache